x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -1615,6 +1618,23 @@ private LogicalPlan rule(Eval eval) {
         }
     }
 
+    /**
+     * Replace type converting eval with aliasing eval when type change does not occur.
+     * A following ReplaceAliasingEvalWithProject will effectively convert ReferenceAttribute into FieldAttribute,
+     * something very useful in local physical planning.
+     */
+    static class ReplaceTypeConvertingEvalWithAliasingEval extends OptimizerRules.OptimizerRule<Eval> {
```

Comment:
nit: name could be a bit more succinct.
```suggestion
    static class ReplaceTrivialTypeConversions extends OptimizerRules.OptimizerRule<Eval> {
```

Reply:
What, but this is _Java_ where unnecessary verbosity is the norm!

OK, seriously though, I like your suggestion. But since the 'suggested change' did not also include the call to this class, I changed it in a separate commit rather than just merging yours.